capstone-rs/src/capstone.rs

```diff
@@ -365,6 +365,44 @@ impl Capstone {
         }
     }
 
+    /// Get the registers are which are read to and written to, in that order.
+    pub fn regs_access(&self, insn: &Insn) -> Option<CsResult<(Vec<RegId>, Vec<RegId>)>> {
```

Comment:
I don't like forcing allocations here by returning `Vec`. You can keep an allocation version for convenience, but please also create a `regs_access_buf()` where the user can provide their own buffers (which may just be stack allocated).

For simplicity, the `regs_access()` can call `regs_access_buf()`

Reply:
Should I just append the registers to the provided buffers, or should i clear them first?

Reply:
If the user may provide stack allocated buffers, doesn't that mean they generally have to provide the length aswell? If they can pass a `regs_read: &mut [u16; MAX_NUM_REGISTERS]`, how will we specify how many were actually read?